pkg/slack/client.go

```diff
@@ -0,0 +1,260 @@
+package enterprise
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/slack-go/slack"
+)
+
+const baseUrl = "https://slack.com/api/"
+
+type Client struct {
+	httpClient   *http.Client
+	token        string
+	enterpriseID string
+	botToken     string
+}
+
+type BaseResponse struct {
+	Ok    bool   `json:"ok"`
+	Error string `json:"error"`
+}
+
+type Pagination struct {
+	ResponseMetadata struct {
+		NextCursor string `json:"next_cursor"`
+	} `json:"response_metadata"`
+}
+
+func NewClient(httpClient *http.Client, token, botToken, enterpriseID string) *Client {
+	return &Client{
+		httpClient:   httpClient,
+		token:        token,
+		enterpriseID: enterpriseID,
+		botToken:     botToken,
+	}
+}
+
+// GetUserInfo returns the user info for the given user ID.
+func (c *Client) GetUserInfo(ctx context.Context, userID string) (*User, error) {
+	values := url.Values{
+		"token": {c.botToken},
+		"user":  {userID},
+	}
+
+	usersUrl, err := url.JoinPath(baseUrl + "users.info")
```

Comment:
Why is this using JoinPath if it's + concatinating the path? What's the expected format of the + vs after the JoinPath? Or is that just to join the URL values?